File: apps/api/src/lunaris_api/schemas/me.py
from .base import CamelModel


class MeResponse(CamelModel):
    """The authenticated caller's identity, plus whether they may manage the signup invite-gate."""

    user_id: str
    # On the admin allowlist (LUNARIS_ADMIN_EMAILS). The web shows the Invitations admin screen only
    # when true; the API still enforces admin access on every admin endpoint regardless.
    is_admin: bool

File: apps/web/src/lib/me.ts
import { authedFetch } from "./apiClient";

/** The signed-in caller's identity, plus whether they may manage the signup invite-gate. */
export interface MeView {
  userId: string;
  isAdmin: boolean;
}

/** Fetch the authenticated caller's identity. Throws on a non-2xx (e.g. 401 when not signed in). */
export async function fetchMe(apiBaseUrl: string, signal?: AbortSignal): Promise<MeView> {
  const response = await authedFetch(
    `${apiBaseUrl}/api/me`,
    signal ? { signal } : undefined,
  );
  if (!response.ok) {
    throw new Error(`Could not load your identity (HTTP ${response.status}).`);
  }
  return (await response.json()) as MeView;
}
